### DISCUSSION TITLE
"driver does not support creating temporary URLs" with image temporaryUrl() + validation

### DISCUSSION THREAD

**Discussion by ahmadeyamin** [livewire/livewire#3133] [7. Bugs (archived)]
### Description
When the user uploads the wrong file format it should prevent $photo to get updated and keep it null or previous value so temporaryUrl() should not work.

but when the user uploads the wrong file format it still changes $photo so in frontend temporaryUrl() gets called and shows an error.

validation should work before $photo update but it works after the update so it shows this kind of error :(

![image](https://user-images.githubusercontent.com/29674341/102015378-7f436500-3d85-11eb-96ff-e678e02dbce6.png)
![image](https://user-images.githubusercontent.com/29674341/102015398-9aae7000-3d85-11eb-8815-35704526e5e5.png)

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: [e.g. 2.3]
- Laravel version: [e.g. 8.*]
- Browser: [e.g. Chrome, Safari]

**Comment by mblarsen** [comment]
Duplicate of https://github.com/livewire/livewire/issues/1106 — see possible workaround in the comments.

**Comment by IsraH** [comment]
Hi, 
go to  livewire.php in config folder of your app,
then add , 'jpg' or another image that can  render on browser at 
'temporary_file_upload' => [
     ...
     'preview_mimes' => [
      ...
      'jpeg', 'jpg', 'mpga', ...
     ...

This solve the issue, this happened to me after upgrade livewire

**Reply by timgavin** [reply]
This does not prevent the `This driver does not support creating temporary URLs` exception from being thrown when selecting a non-image file.

**Comment by klesou** [comment]
Hi,

I was thinking that maybe it's a context problem. 

There is no need to use the temporaryUrl function if you not expect an image, which can be previewed.

But, sometimes, we click in the wrong file and then, there is a crash. 

What lead us to deal with this problem in the front and maybe with an ugly solution.

And I thought that we could display an image in base64 that inform us that this file can not be previewed. 

So, the solution that I put in my project was to insert this in line 86 of TemporaryUploadedFile.php:


if( ! $this->isPreviewable() )
{
 return "data:image/png;base64...gg=="; // base of the image that shows that is not a previewed file'*/
}


For me it work's, both in code and in thoughts process, because dont break de flow and show to the user that there is something wrong with the file. "Hey, it was supossed to be an image". And then, when the user submit the form without change the file, the validation will be triggered.

What you guys, think about it?

**Reply by boedyirh** [reply]
I think you should submit a PR for this.

**Comment by timgavin** [comment]
@klesou Livewire is basically unusable for file uploads because of this stupid (and old!) `temporaryUrl` bug. And every workaround I've come across still crashes the app. Yours is the only solution that actually works consistently. The only thing it's missing is an error message, but since these files pass validation what can you do...

I put it on line 78, right at the top of the `temporaryUrl()` method and it's working perfectly. Thank you.

You should submit a PR and get this adopted. :)

```php
public function temporaryUrl()
{
    if (! $this->isPreviewable()) {
        // show a missing image icon (?) for files that cannot be previewed
        return 'data:image/png;base64...gg==';
    }
    ...
}

**Reply by timgavin** [reply]
Unfortunately this solution is really only a band-aid, as the real problem is the validator allowing non-images to pass `image` validation during an `update`

**Comment by osama-98** [comment]
# Solution: Reset file variables after form submission
I encountered this same issue and spent considerable time debugging it. The solution that worked for me was surprisingly simple: reset the file variable after processing the form submission.

the solution was only to reset the image variable:
```php
$this->reset(['image']);
```

## Simple Example


```php
<?php

use Livewire\Component;
use Livewire\WithFileUploads;

class ImageUpload extends Component
{
    use WithFileUploads;

    public  $image = null;

    protected function rules(): array
    {
        return [
            'image' => 'required|image|max:4096',
        ];
    }

    public function save(): void
    {
        $this->validate();

        // Process your file upload logic here
        // For example: store the file, save to database, etc.
        
        // IMPORTANT: Reset the file variable after processing
        // This prevents temporary URL issues on subsequent uploads
        $this->reset(['image']);
        
        session()->flash('message', 'Image uploaded successfully!');
    }

    public function render()
    {
        return view('livewire.image-upload');
    }
}
```
